src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -63,6 +64,10 @@ public Post createPost(Long courseId, Post post) {
 
         // set author to current user
         post.setAuthor(user);
+
+        // set creation date to now
+        post.setCreationDate(ZonedDateTime.now());
```

Comment:
I see this added in multiple places. Since the date is now(). What if we were to set the date automatically upon entity **save()** like we do [here](url) and [here](https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/domain/analytics/TextAssessmentEvent.java#L33). 
For example in _Posting.java_ we could have something like this: 
```
    @Column(name = "creation_date")
    private ZonedDateTime creationDate = ZonedDateTime.now();
```
then when you do `postRepository.save(post);` on this entity, date is set automatically.

Reply:
I changed it, thanks for the suggestion